runtime/src/bank.rs

```diff
@@ -3805,7 +3804,7 @@ impl Bank {
             .stakes_cache
             .stakes()
             .highest_staked_node()
-            .unwrap_or_default();
+            .unwrap_or_else(Pubkey::new_unique);
```

Comment:
this method should absolutely not be used outside of tests under any circumstances.  can we just fail/panic instead?

Reply:
It's only used in tests right now... using the system program (default pubkey) as the highest staked node should also not happen outside of tests in any circumstances. I'll clean it up in a follow up change though to be sure

Reply:
PR fix is being worked on here: https://github.com/solana-labs/solana/pull/34135